junit-platform-commons/src/main/java/org/junit/platform/commons/util/ExceptionUtils.java

```diff
@@ -80,4 +94,77 @@ public static String readStackTrace(Throwable throwable) {
 		return stringWriter.toString();
 	}
 
+	/**
+	 * Prune the stack trace of the supplied {@link Throwable} by filtering its
+	 * elements using the supplied {@link Predicate}, except for
+	 * {@code org.junit.jupiter.api.Assertions} and
+	 * {@code org.junit.jupiter.api.Assumptions} that will always remain
+	 * present.
+	 *
+	 * <p>Additionally, all elements prior to and including the first
+	 * JUnit Launcher call will be removed.
+	 *
+	 * @param throwable the {@code Throwable} whose stack trace should be
+	 * pruned; never {@code null}
+	 * @param stackTraceElementFilter the {@code Predicate} used to filter
+	 * elements of the stack trace; never {@code null}
+	 *
+	 * @since 5.10
+	 */
+	public static void pruneStackTrace(Throwable throwable, Predicate<String> stackTraceElementFilter) {
```

Comment:
```suggestion
	@API(status = INTERNAL, since = "1.10")
	public static void pruneStackTrace(Throwable throwable, Predicate<String> stackTraceElementFilter) {
```